### PR TITLE
[WIP] Make a system for keeping docs and file names in sync

### LINKED ISSUE #78: AMICO config should probably get some sort of different name
Note to self: This is a config file and should probably get some sort of different name

_Originally posted by @mattcieslak in https://github.com/PennLINC/qsirecon/pull/72#discussion_r1723727704_

### PR DESCRIPTION
**Pull request by mattcieslak**
The recon scalars are a special case in that they're shared across nodes and we have a good system for documenting them. But the other outputs from each node are currently just being sent to regular datasinks with the bids entities hard coded into the workflow code.

This PR sets up a new directory of yaml files for nonscalar outputs and their bids entities / metadata. Since BEP016 doesn't cover these, there will likely be significant changes in the file names, so having something flexible is going to be great. 

Comments would be appreciated!

### CODE REVIEW

File: qsirecon/workflows/recon/amico.py [hunk 1]
```diff
@@ -141,12 +143,12 @@ def init_amico_noddi_fit_wf(
         ])  # fmt:skip
 
     if qsirecon_suffix:
+        derivatives_config = load_yaml(load_data("scalars/amico_noddi.yaml"))
```
Comment (tsalo) [inline qsirecon/workflows/recon/amico.py:146]: I think this should be `nonscalars`

**Comment by mattcieslak** [comment]
Add paths to yaml files to docstrings of workflows

**Comment by tsalo** [comment]
It looks like the problem is that you're trying to resolve #78 here, but you need to change the io_spec.json file to make that work.

**Comment by mattcieslak** [comment]
Alright, this looks pretty good in the docs. I'm going to update it for the rest of the workflows too

**Pull request merged**